### DISCUSSION TITLE
Seeing Entry point '@clr/angular' contains deep imports

### DISCUSSION THREAD

**Discussion by kumar-tadepalli** [vmware/clarity#6007] [Q&A]
Hi Team,

I'm seeing this error while building and my project uses clarity angular only and i'm not using cds/core not even imported in app module it is just present in package.json.

But Im seeing the below warnings. Can you provide any info, whether i'm missing anything?

Warning: Entry point '@clr/angular' contains deep imports into 'C:/Users/administrator/Desktop/ng11/app3/node_modules/@cds/core/icon', 'C:/Users/administrator/Desktop/ng11/app3/node_modules/@cds/core/icon/register'. This is probably not a problem, but may cause the compilation of entry points to be out of order.
Warning: Entry point '@cds/angular' contains deep imports into 'C:/Users/administrator/Desktop/ng11/app3/node_modules/@cds/core/accordion/register.js', 'C:/Users/administrator/Desktop/ng11/app3/node_modules/@cds/core/alert/register.js', 'C:/Users/administrator/Desktop/ng11/app3/node_modules/@cds/core/badge/register.js', 'C:/Users/administrator/Desktop/ng11/app3/node_modules/@cds/core/button/register.js', 'C:/Users/administrator/Desktop/ng11/app3/node_modules/@cds/core/checkbox/register.js', 'C:/Users/administrator/Desktop/ng11/app3/node_modules/@cds/core/datalist/register.js', 'C:/Users/administrator/Desktop/ng11/app3/node_modules/@cds/core/date/register.js', 'C:/Users/administrator/Desktop/ng11/app3/node_modules/@cds/core/divider/register.js', 'C:/Users/administrator/Desktop/ng11/app3/node_modules/@cds/core/file/register.js', 'C:/Users/administrator/Desktop/ng11/app3/node_modules/@cds/core/forms/register.js', 'C:/Users/administrator/Desktop/ng11/app3/node_modules/@cds/core/icon/register.js', 'C:/Users/administrator/Desktop/ng11/app3/node_modules/@cds/core/input/register.js', 'C:/Users/administrator/Desktop/ng11/app3/node_modules/@cds/core/modal/register.js', 'C:/Users/administrator/Desktop/ng11/app3/node_modules/@cds/core/password/register.js', 'C:/Users/administrator/Desktop/ng11/app3/node_modules/@cds/core/progress-circle/register.js', 'C:/Users/administrator/Desktop/ng11/app3/node_modules/@cds/core/radio/register.js', 'C:/Users/administrator/Desktop/ng11/app3/node_modules/@cds/core/range/register.js', 'C:/Users/administrator/Desktop/ng11/app3/node_modules/@cds/core/search/register.js', 'C:/Users/administrator/Desktop/ng11/app3/node_modules/@cds/core/select/register.js', 'C:/Users/administrator/Desktop/ng11/app3/node_modules/@cds/core/tag/register.js', 'C:/Users/administrator/Desktop/ng11/app3/node_modules/@cds/core/test-dropdown/register.js', 'C:/Users/administrator/Desktop/ng11/app3/node_modules/@cds/core/textarea/register.js', 'C:/Users/administrator/Desktop/ng11/app3/node_modules/@cds/core/time/register.js', 'C:/Users/administrator/Desktop/ng11/app3/node_modules/@cds/core/toggle/register.js'. This is probably not a problem, but may cause the compilation of entry points to be out of order.

**Comment by kumar-tadepalli** [comment]
Any update on this issue?

**Comment by Jinnie** [comment]
These warnings are issued from Angular ngcc compiler. As commented here: https://github.com/angular/angular/issues/35615#issuecomment-590558673 
> These are only warnings. Given that the devextreme package is not compiled by Angular then, as pointed out in the message: "This is probably not a problem."
>
> It is not possible for ngcc to know whether the deep imports are going to be problematic or not, so it is reasonable to give the developer this information so that they can make a decision for themselves.

Now why do we have them in Clarity Core? 
They are intentionally used as part of the API that allows importing only the code that you will really need, not the whole framework at once. So you can safely ignore the warnings.
